### PR TITLE
Changed Run task to show instructions on how to run the demo - issue #98

### PR DESCRIPTION
**Pull request by thrandale**
Provides a solution for https://github.com/MetroCS/redistricting/issues/98

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -148,7 +148,9 @@
 
   <target name="run" depends="jar"
           description="Runs the development version of the program">
```
Comment (jody) [inline build.xml:150]: This description is no longer appropriate to this task. Please revise to match.

File: build.xml [hunk 2]
```diff
@@ -148,7 +148,9 @@
...
-    <java jar="${jar.dir}/${productname}.jar" fork="true"/>
+          <echo message="To run the demonstration:"/>
+          <echo message="Build using 'ant jar'"/>
+          <echo message="Run using 'java -jar build\jar\Redistricting.jar'"/>
```
Comment (jody) [inline build.xml:153]: Please construct this comment dynamically based on the configured directory and product name. (You can use the previous java task definition as a template.)

**Review by jody — CHANGES_REQUESTED** [review]

File: build.xml (hunk 1 above)
Comment (jody) [reply build.xml]: Description still needs to be changed, so that it doesn't claim to run the development version of the program.

Comment (thrandale) [reply build.xml]: All requested changes have been made, please review

**Review by jody — CHANGES_REQUESTED** [review]
Awaiting response to question regarding "Instructions" string.

File: build.xml [hunk 3]
```diff
@@ -147,8 +147,10 @@
   </target>
 
   <target name="run" depends="jar"
```
Comment (jody) [inline build.xml:149]: See comment regarding line 152 that specifies running "ant jar".

File: build.xml [hunk 4]
```diff
@@ -147,8 +147,10 @@
...
-          description="Runs the development version of the program">
-    <java jar="${jar.dir}/${productname}.jar" fork="true"/>
+          description="Gives Instructions on how to run">
+          <echo message="To run the demonstration:"/>
+          <echo message="Build using 'ant jar'"/>
```
Comment (jody) [inline build.xml:152]: There seems to be a mismatch among the description, dependencies, and echoed messages.

If "run" depends on "jar" (as specified in line 149) then why should invoking the "jar" target be part of the instructions given by the "run" target? Using "ant run" would automatically first run "jar" (if necessary) and then show the messages.

**Review by jody — CHANGES_REQUESTED** [review]

File: build.xml [hunk 5]
```diff
@@ -147,8 +147,10 @@
   </target>
 
   <target name="run" depends="jar"
-          description="Runs the development version of the program">
-    <java jar="${jar.dir}/${productname}.jar" fork="true"/>
+          description="Gives Instructions on how to run">
```
[the remark this answers is not in the document]
Comment (thrandale) [reply build.xml:150]: Just a proofreading error.

File: build.xml [hunk 6]
```diff
@@ -147,10 +147,10 @@
   </target>
 
   <target name="run" depends="jar"
-          description="Gives Instructions on how to run">
+          description="Gives instructions on how to run">
           <echo message="To run the demonstration:"/>
-          <echo message="Build using 'ant jar'"/>
-          <echo message="Run using 'java -jar ${jar.dir}\${productname}.jar'"/>
+          <echo message="Build using 'ant Run'"/>
```
Comment (jody) [inline build.xml:152]: This appears to be circular (running the "run" task now advises running the "Run" task) unless capitalization is significant and a new task named "Run" had been created. (Which isn't the intent.)

**Review by jody — CHANGES_REQUESTED** [review]

File: build.xml [hunk 7]
```diff
@@ -147,8 +147,10 @@
   </target>
 
   <target name="run" depends="jar"
-          description="Runs the development version of the program">
-    <java jar="${jar.dir}/${productname}.jar" fork="true"/>
+          description="Gives instructions on how to run">
+          <echo message="To run the demonstration:"/>
+          <echo message="Build using 'ant run'"/>
```
Comment (jody) [inline build.xml:152]: This self-referential instruction results in the command `ant run` telling the user to "Build using 'ant run'".

**Review by jody — CHANGES_REQUESTED** [review]

**Comment by thrandale** [comment]
We have changed the run task to no longer depend on the "jar" task. and we direct the user to use "ant jar" to build it.

**Review by jody — COMMENTED** [review]
Removing RedistrictingDriver from the package is becoming a highly likely outcome. All issues/PRs that address concerns involving RedistrictingDriver are thus likely to be irrelevant. Am putting all such activity "on hold" for now.

**Comment by jody** [comment]
See new issue #145

**Pull request closed without merging**